enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,276 @@
+
+  - The Smart NICs could be part of a second cluster, where the
+    control planes for both clusters are running in a different
+    cluster, ROKS-style.
+
+The latter two approaches allow for more isolation between the hosts
+and the NICs, since potentially even the cluster administrators of the
+primary cluster might not be able to influence the NIC cluster. This
+would be especially important if additional primary-cluster security
+and monitoring functions were offloaded to the NICs in the future.
+
+### Initial Development With Beta BlueField-2 Hardware
+
+Although we will eventually want to use IPMI to control and provision
+the NICs, just like we do for the hosts with bare-metal IPI, this will
+not work with the pre-GA BlueField-2 cards that we currently have
+access to. Instead, we will have to communicate with the NICs over PCI
+from the host. The most future-compatible approach is probably to
+create an IPMI proxy that can run on each host, claiming its own
+separate IP address on the provisioning network, and then proxying
+commands to the NIC via the PCI "RShim" interface. (Thus, we will have
+to first install OCP on the node, then deploy a DaemonSet to the node
+to run the IPMI proxy, then install OCP on the NIC via the proxy.)
+
+Later on, it should be possible to remove the proxy and instead speak
+IPMI directly to the NIC's Baseboard Management Controller connected
+directly via the out-of-band BMC network port on the card.
+
+(One unresolved problem with the "RShim" approach is that based on the
+proof of concept prototypes, it does not seem to be possible to do a
+fully-unsupervised installation via that interface. At one point, user
+intervention over a serial console is required to pick the correct
+option from a boot menu whose contents cannot be reliably predicted in
+advance. We may have to just have the user ssh in at some point during
+the install process to deal with this.)
+
+This would explicitly be a temporary hack which would not ever be part
+of a released version of OCP.
+
+### ...
+
+...
+
+## Risks and Mitigations
+
+### External Schedule/Feature Changes
+
+We cannot commit to shipping this feature in any specific OCP release
+at this time, since it depends on when NVIDIA releases the GA version
+of the BlueField-2 card.
+
+We are currently developing using pre-GA hardware, with a plan based
+on official descriptions of how the final GA BlueField-2 hardware will
+work. In the event that the product is changed before GA, some or all
+of our plans may need to be revisited.
+
+At the time of writing, the network driver for the out-of-band BMC
+network port has been posted to LKML and gone through some review, but
+has not yet merged to the Linux kernel. We will need to have this
+driver in the (ARM) RHCOS kernel to be able to do an OCP install over
+the BMC port. While we assume that this driver will merge upstream
+before the product goes GA, it is possible that it won't.
```

Comment:
IMO it's a hard requirement.

Reply:
Sorry, "the product" in that sentence was referring to BlueField-2, not OCP. ie, It's possible that NVIDIA will ship the GA cards before they've finished upstreaming this driver, in which case we have to figure out what we want to do.